WooCommerce/UITestsFoundation/Screens/Orders/NewOrderScreen.swift

```diff
@@ -58,6 +66,14 @@ public final class NewOrderScreen: ScreenObject {
         return try AddProductScreen()
     }
 
+    /// Opens the Customer Details screen.
+    /// - Returns: Customer Details screen object.
+    @discardableResult
+    public func openCustomerDetailsScreen() throws -> CustomerDetailsScreen {
+        addCustomerDetailsButton.tap()
```

Comment:
Since the main purpose of the `init` is to complete that implicit assertion, I generally add the most unique or important element on a particular screen to the `init`. Does that make sense?
Especially in cases where the other elements will be used in the test, it seems like adding every element we plan to interact with here could slow down the test, and I'm not sure it gives us more information if the test fails.
The NewOrderScreen in particular has a lot of elements on it, so I'm a little hesitant to go this route. I may be missing some reason to check for elements' presence before using them though.

Reply:
> Since the main purpose of the init is to complete that implicit assertion, I generally add the most unique or important element on a particular screen to the init. Does that make sense?

Yes, this makes perfect sense, and you're right about the time factor. Addition of one element into the `init` slows down this test alone by 5 seconds. I have to admit that I did not test my suggestion before writing the first comment, and it actually won't work: the "Add Customer Details" button will become absent on the screen after the details were entered. So, this extended init will work on the first screen load, but will fail once we enter the customer details, submit, and go back to the screen - the button will simply not be there any more.

> I may be missing some reason to check for elements' presence before using them though.

Usually the fact of element absence causes the test to stop immediately (because we have `continueAfterFailure = false`), and the test runner might provide some useful logging, like this in our case: `Failed to get matching snapshot: No matches found for Elements matching predicate '"Add Customer Details" IN identifiers' from input...`.

However, in some cases you might want to have more control, and not stop the test, but adapt the test scenario depending on the elements' presence, or just have some custom logging. It's not widespread, but I imagine the Contact Support test being such an example, where we can have the same test for both cases of ZD API key present and absent.

I suspect my suggestion and habit of checking the elements' presence before using them comes from the other testing frameworks, where the default logging and test flow control was not as good as in XCTest.